crates/pop-parachains/src/build.rs

```diff
@@ -1,13 +1,498 @@
 // SPDX-License-Identifier: GPL-3.0
 
+use crate::Error;
+use anyhow::Result;
 use duct::cmd;
-use std::path::PathBuf;
+use pop_common::replace_in_file;
+use serde_json::Value;
+use std::{
+	collections::HashMap,
+	fs,
+	path::{Path, PathBuf},
+};
+use toml_edit::DocumentMut;
 
-/// Build the parachain located in the specified `path`.
-pub fn build_parachain(path: &Option<PathBuf>) -> anyhow::Result<()> {
+/// Build the parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+pub fn build_parachain(path: &Option<PathBuf>) -> Result<(), Error> {
 	cmd("cargo", vec!["build", "--release"])
 		.dir(path.clone().unwrap_or("./".into()))
 		.run()?;
+	Ok(())
+}
+
+/// Constructs the node binary path based on the target path and the node folder path.
+///
+/// # Arguments
+/// * `target_path` - The path where the binaries are expected to be found.
+/// * `node_path` - The path to the node from which the node name will be parsed.
+pub fn binary_path(target_path: &Path, node_path: &Path) -> Result<PathBuf, Error> {
```

Comment:
As suggested elsewhere, this can be eliminated by including the logic in build_parachain to return the built binary path, which is much simpler for the caller. Fine to keep it as a separate private function for testing if you want, its just the current usage is unnecessarily noisy.

Reply:
removed the pub! and use only internally